### PR TITLE
Add support for solc path

### PR DESCRIPTION
**Pull request by cehali**
Added support for solc path, which can be passed to solcVersion in waffle.json.

### CODE REVIEW

File: test/utils.ts [hunk 1]
```diff
@@ -15,4 +15,18 @@ describe('UNIT: Utils', () => {
       expect(flattenObjectArray([{ab: 1}, {cb: 2}, {dd: 4}])).to.deep.eq({ab: 1, cb: 2, dd: 4});
     });
   });
+
+  describe('isDirectory', () => {
```
Comment (sz-piotr) [inline test/utils.ts]: These are not unit tests

File: test/compiler/wrappers/compileSolcjs.ts [hunk 2]
```diff
@@ -17,3 +17,20 @@ describe('UNIT: findInputs', () => {
     expect(basicTokenContractActual).to.deep.eq(basicTokenContractExpected);
   });
 });
+
+describe('UNIT: loadCompiler', () => {
```
Comment (sz-piotr) [inline test/compiler/wrappers/compileSolcjs.ts]: Not unit tests

File: test/compiler/wrappers/compileSolcjs.ts [hunk 3]
```diff
@@ -17,3 +17,20 @@ describe('UNIT: findInputs', () => {
...
+  it('loadCompiler with solcVersion as version', async () => {
+    const solc = await loadCompiler({solcVersion: 'v0.5.9+commit.e560f70d', sourcesPath: 'test', npmPath: 'test'});
+    expect(solc.version()).to.eq('0.5.9+commit.e560f70d.Emscripten.clang');
+  });
+
+  it('loadCompiler with solcVersion as path', async () => {
+    const solc = await loadCompiler({solcVersion: 'node_modules/solc', sourcesPath: 'test', npmPath: 'test'});
+    expect(solc.version()).to.eq('0.5.11+commit.c082d0b4.Emscripten.clang');
```
Comment (sz-piotr) [inline test/compiler/wrappers/compileSolcjs.ts]: obtain this using `require('solc')`

**Comment by sz-piotr** [comment]
Document the change

File: docs/source/configuration.rst [hunk 4]
```diff
@@ -50,6 +50,14 @@ Solcjs allows switching used version of solidity compiler on the fly. To set up
...
 Version naming is somewhat counter-intuitive. You can deduce version name from `list available here <https://ethereum.github.io/solc-bin/bin/list.json>`_.
 
+Also Solcjs can be provided by passing path to Solcjs directory. Path have to be relative to working directory.
+::
+  
+  {
+    ...
+    "solcVersion": "./node_modules/solc"
+  }
+
```
Comment (marekkirejczyk) [inline docs/source/configuration.rst]: 👍

File: lib/compiler/compileSolcjs.ts [hunk 5]
```diff
@@ -1,18 +1,20 @@
...
 import { ImportFile } from '@resolver-engine/imports';
 
 const loadRemoteVersion = promisify(solc.loadRemoteVersion);
 
-async function loadCompiler(config: Config) {
+export async function loadCompiler(config: Config) {
   if (config.solcVersion) {
+    if (isDirectory(config.solcVersion)) {
+      return require(relativePathToWorkingDir(config.solcVersion));
+    }
```
Comment (marekkirejczyk) [inline lib/compiler/compileSolcjs.ts]: throw error otherwiee

File: test/compiler/wrappers/compileSolcjs.ts [hunk 6]
```diff
@@ -1,14 +1,19 @@
...
   'test/projects/example/ERC20Basic.sol',
   'test/projects/example/mock/BasicTokenMock.sol'
 ];
 
-describe('UNIT: findInputs', () => {
+const requiredInputs = {
+  sourcesPath: 'test', npmPath: 'test'
+};
+
+describe('INTEGRITY: findInputs', () => {
```
Comment (marekkirejczyk) [inline test/compiler/wrappers/compileSolcjs.ts]: 'INTEGRATION'

File: test/compiler/wrappers/compileSolcjs.ts [hunk 7]
```diff
@@ -17,3 +22,20 @@ describe('UNIT: findInputs', () => {
     expect(basicTokenContractActual).to.deep.eq(basicTokenContractExpected);
   });
 });
+
+describe('INTEGRITY: loadCompiler', () => {
```
Comment (marekkirejczyk) [inline test/compiler/wrappers/compileSolcjs.ts]: 'INTEGRATION'

File: test/utils.ts [hunk 8]
```diff
@@ -15,4 +15,18 @@ describe('UNIT: Utils', () => {
       expect(flattenObjectArray([{ab: 1}, {cb: 2}, {dd: 4}])).to.deep.eq({ab: 1, cb: 2, dd: 4});
     });
   });
+
+  describe('INTEGRITY: isDirectory', () => {
```
Comment (marekkirejczyk) [inline test/utils.ts]: 'INTEGRATION'

File: yarn.lock [hunk 9]
```diff
@@ -233,7 +233,7 @@ ansi-styles@^2.2.1:
   version "2.2.1"
   resolved "https://registry.yarnpkg.com/ansi-styles/-/ansi-styles-2.2.1.tgz#b432dd3358b634cf75e1e4664368240533c1ddbe"
 
-ansi-styles@^3.2.1:
+ansi-styles@^3.2.0, ansi-styles@^3.2.1:
```
Comment (marekkirejczyk) [inline yarn.lock]: Why `yarn.lock` has changed? Accident? On purpose -> separate PR, please.

**Pull request merged**